### ISSUE TITLE
Let the user decide the beginning of the interval

### ISSUE THREAD

**Issue by joeldomke** [imaNNeoFighT/fl_chart#893]
When I choose an intervall of 50, I always want my titles to be at 0, 50, 100, ... This is what happens, if I set my minY to a multiple of 50, or if 0 is between minY and maxY.

But if I select something like minY = 35 and maxY = 130, I will get titles at 35 and 85.

The user should have more control over the beginning of the interval. I would like to implement this feature, and there are two solutions that come to my mind:

1. Give an option to have the interval zero aligned.
2. Let the user choose a value at which the interval should be aligned. E.g. the user choses 10 for the alignment, 50 as the interval, 30 as minY and 150 as maxY, then they would get titles at 60 and 110.

This could be either set for horizontal and vertical, or for left, top, right and bottom.

**Comment by joeldomke** [comment]
I just noticed that the interval is also used for the grid lines. So this change should probably also apply to the grid lines, since they are currently placed on the same interval as the titles.

**Comment by imaNNeoFighT** [comment]
Yes, you're right!

**Comment by imaNNeoFighT** [comment]
At the moment [we have a function](https://github.com/imaNNeoFighT/fl_chart/blob/860a00b5d6d9b4c11864775882d7987f8d98a48a/lib/src/utils/utils.dart#L262) called `Utils.getBestInitialIntervalValue(min, max, interval)` 
with this definition:
```dart
/// Finds the best initial interval value
///
/// If there is a zero point in the axis, we a value that passes through it.
/// For example if we have -3 to +3, with interval 2. if we start from -3, we get something like this: -3, -1, +1, +3
/// But the most important point is zero in most cases. with this logic we get this: -2, 0, 2
double getBestInitialIntervalValue(double min, double max, double interval) {
  if (min > 0 || max < 0) {
    return min;
  }
  if (max - min <= interval) {
    return min;
  }
  return interval * (min ~/ interval).toDouble();
}
  ```

It tries to find a value as an initial value based on the interval, we wanted to make sure that this value goes through the zero when zero is between in `min` and `max`. But it didn't cover your sample (35 to 130) because there is no zero in between :))

Maybe you can work on this function to fix this issue without adding any parameters.

BTW your second solution makes sense to me.

**Comment by joeldomke** [comment]
Without changing the parameters there isn't much flexibility. Option 1 would be possible with the limitation that we can't make it optional (if I am not mistaken). Meaning it would change the behavior of existing charts, which would have to be mentioned as a breaking change.

In this case we could remove the if conditions and only use
```dart
 return interval * (min ~/ interval).toDouble();
```
to always have the interval zero aligned.

**Comment by imaNNeoFighT** [comment]
@joeldomke You're right we can add that option.
That function is implemented with a predefined value for interval for zero (there is no parameter, it was just on my head).
Now if we can fix the problem for zero, we can then make it an optional value.

I'm working on it right now.

**Comment by imaNNeoFighT** [comment]
What is your suggestion for name of that optional value? `baselineAlignment`?
(which the default value is zero)

**Comment by imaNNeoFighT** [comment]
Actually, I'm thinking of adding a `baseline` property in our axis-based charts (LineChart, BarChart, ...)

The default value of the baseline is zero in both axes.
Then we can show the base lines attractive in the grid lines or even the titles by default.
Something like the below image:
<img src="https://user-images.githubusercontent.com/7009300/152528433-efbba392-065f-4471-a5d7-f3be98ffcb8b.png" width=400 />

Then we can use the `baseline` value to align the interval that we are talking about.

What do you think?

**Comment by joeldomke** [comment]
I think that's a great idea!

**Comment by imaNNeoFighT** [comment]
‍> Let the user choose a value at which the interval should be aligned. E.g. the user choses 10 for the alignment, 50 as the interval, 30 as minY and 150 as maxY, then they would get titles at 60 and 110.

This way, it should cover your example. (with baseline `10`).

Correct?

**Comment by joeldomke** [comment]
> This way, it should cover your example. (with baseline `10`).

Yes, this would fix it

**Comment by imaNNeoFighT** [comment]
https://user-images.githubusercontent.com/7009300/152555425-3b53ac8c-257f-49b0-8d75-1a878c03ccaa.mp4

**Comment by imaNNeoFighT** [comment]
Hi. 
We have just fixed it in [0.45.0](https://github.com/imaNNeoFighT/fl_chart/releases/tag/0.45.0).
Please check it out!

**Comment by joeldomke** [comment]
Looks good, thanks!
